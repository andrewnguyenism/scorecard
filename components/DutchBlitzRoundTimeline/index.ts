export { DutchBlitzRoundTimeline } from './DutchBlitzRoundTimeline'

export type { DutchBlitzRound } from './DutchBlitzRoundTimeline'
