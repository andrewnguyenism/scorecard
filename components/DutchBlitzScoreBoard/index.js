export {
  DutchBlitzScoreBoardHeader,
  DutchBlitzScoreBoardPlayer,
  DutchBlitzScoreBoard,
} from "./DutchBlitzScoreBoard";
