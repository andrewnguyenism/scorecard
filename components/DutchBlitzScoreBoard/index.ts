export { DutchBlitzScoreBoard } from './DutchBlitzScoreBoard'
export { DutchBlitzScoreBoardHeader } from './DutchBlitzScoreBoardHeader'
export { DutchBlitzScoreBoardPlayer } from './DutchBlitzScoreBoardPlayer'
