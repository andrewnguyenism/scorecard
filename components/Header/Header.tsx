import { FunctionComponent } from 'react'

import Link from 'next/link'
import { useRouter } from 'next/router'

export const Header: FunctionComponent = () => {
  const router = useRouter()
  return (
    <div className="flex items-center bg-gray-700 text-white justify-between sha h-16 px-4">
      {router.asPath !== '/' ? (
        <Link href="/">
          <div className="p-4 -ml-4">
            <svg
              className="h-6 w-6"
              xmlns="http://www.w3.org/2000/svg"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M10 19l-7-7m0 0l7-7m-7 7h18"
              />
            </svg>
          </div>
        </Link>
      ) : (
        <div className="w-6"></div>
      )}
      <span className="text-4xl font-bold font-mono">ScoreCard</span>
      <div className="w-6"></div>
    </div>
  )
}
