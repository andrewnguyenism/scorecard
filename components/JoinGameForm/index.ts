export { JoinGameFormError, JoinGameForm } from './JoinGameForm'
