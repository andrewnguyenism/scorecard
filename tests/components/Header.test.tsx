import { render } from '../test-utils'

import { Header } from '@/components/Header'

describe('Header component', () => {
  it('renders without back button', () => {
    const { asFragment } = render(<Header />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders with back button', () => {
    const { asFragment } = render(<Header />, {
      router: { asPath: '/new-game/dutch-blitz' },
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
