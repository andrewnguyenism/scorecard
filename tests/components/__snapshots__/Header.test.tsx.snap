// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders with back button 1`] = `
<DocumentFragment>
  <div
    class="flex items-center bg-gray-700 text-white justify-between sha h-16 px-4"
  >
    <div
      class="p-4 -ml-4"
    >
      <svg
        class="h-6 w-6"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 19l-7-7m0 0l7-7m-7 7h18"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
    <span
      class="text-4xl font-bold font-mono"
    >
      ScoreCard
    </span>
    <div
      class="w-6"
    />
  </div>
</DocumentFragment>
`;

exports[`Header component renders without back button 1`] = `
<DocumentFragment>
  <div
    class="flex items-center bg-gray-700 text-white justify-between sha h-16 px-4"
  >
    <div
      class="w-6"
    />
    <span
      class="text-4xl font-bold font-mono"
    >
      ScoreCard
    </span>
    <div
      class="w-6"
    />
  </div>
</DocumentFragment>
`;
